### PR TITLE
Update go version & add verification/testing tools

### PR DESCRIPTION
**Pull request by apoorvajagtap**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Run verifications and test

- [ ] `make verify` is passing
- [ ] `make test` is passing

### CODE REVIEW

File: v2/json2/testapp/counter.go [hunk 1]
```diff
@@ -43,8 +44,17 @@ func main() {
...
+	if err != nil {
+		log.Fatal(err)
+	}
 	http.Handle("/", http.StripPrefix("/", http.FileServer(http.Dir("./"))))
 	http.Handle("/jsonrpc/", s)
-	log.Fatal(http.ListenAndServe(*address, nil))
+
+	server := &http.Server{
+		Addr:              *address,
+		ReadHeaderTimeout: 5 * time.Second,
```
Comment (apoorvajagtap) [inline v2/json2/testapp/counter.go:56]: What should  be the efficient timeout?

File: json/client.go [hunk 2]
```diff
@@ -37,10 +40,15 @@ type clientResponse struct {
 
 // EncodeClientRequest encodes parameters for a JSON-RPC client request.
 func EncodeClientRequest(method string, args interface{}) ([]byte, error) {
+	val, err := rand.Int(rand.Reader, big.NewInt(int64(math.MaxInt64)))
```
Comment (apoorvajagtap) [inline json/client.go:43]: @coreydaley this section needs special attention during review.

Comment (coreydaley) [reply json/client.go:43]: I am working on a similar issue with the websockets repository, swapping math/rand for crypto/rand, haven't gotten it completely working yet though: https://github.com/gorilla/websocket/pull/840/files#diff-4f427d2b022907c552328e63f137561f6de92396d7a6e8f6c2ea1bcf0db52654R186-R194

File: v2/json2/testapp/counter.go (hunk 1 above)
Comment (coreydaley) [reply v2/json2/testapp/counter.go:56]: I think that 5 minutes is reasonable.

File: json/client.go (hunk 2 above)
Comment (coreydaley) [reply json/client.go:43]: Ah, looks like you don't have a weird test that goes with it though :) 
I think this looks fine, you are really just changing it to an `actual` random number :)
https://onlinegdb.com/k7-YE1UhC

File: json/server.go [hunk 3]
```diff
@@ -69,7 +70,10 @@ func newCodecRequest(r *http.Request) rpc.CodecRequest {
 	// Decode the request body and check if RPC method is valid.
 	req := new(serverRequest)
 	err := json.NewDecoder(r.Body).Decode(req)
-	r.Body.Close()
+	er := r.Body.Close()
+	if er != nil {
+		log.Print(er)
+	}
```
Comment (coreydaley) [inline json/server.go:76]: I think that this should close itself

File: v2/json2/server.go [hunk 4]
```diff
@@ -7,12 +7,13 @@ package json2
...
 import (
 	"encoding/json"
+	"log"
 	"net/http"
 
 	"github.com/gorilla/rpc/v2"
 )
 
-var null = json.RawMessage([]byte("null"))
+// var null = json.RawMessage([]byte("null"))
```
Comment (coreydaley) [inline v2/json2/server.go:16]: Can you give some context on this? Is it not being used anywhere?

File: v2/json2/server.go [hunk 5]
```diff
@@ -117,7 +118,10 @@ func newCodecRequest(r *http.Request, encoder rpc.Encoder, errorMapper func(erro
 		}
 	}
 
-	r.Body.Close()
+	er := r.Body.Close()
+	if er != nil {
+		log.Print(er)
+	}
```
Comment (coreydaley) [inline v2/json2/server.go:124]: I think this should close itself

**Review by coreydaley — CHANGES_REQUESTED** [review]

File: json/client.go (hunk 2 above)
Comment (apoorvajagtap) [reply json/client.go:43]: So, it's good to proceed as it is, right?

File: json/json_test.go [hunk 6]
```diff
@@ -83,7 +83,10 @@ func executeRaw(t *testing.T, s *rpc.Server, req interface{}, res interface{}) i
 func TestService(t *testing.T) {
 	s := rpc.NewServer()
 	s.RegisterCodec(NewCodec(), "application/json")
-	s.RegisterService(new(Service1), "")
+	err := s.RegisterService(new(Service1), "")
+	if err != nil {
+		t.Fatal(err)
+	}
```
[the remark this answers is not in the document]
Comment (apoorvajagtap) [reply json/json_test.go]: That actually sounds better. I don't have a problem updating such pieces to make them more readable. On it!

File: v2/json2/server.go (hunk 4 above)
Comment (apoorvajagtap) [reply v2/json2/server.go:16]: Yes, that type is nowhere used in v2/json2/server.go.
Some examples of it being used were found specifically in [WriteResponse](https://github.com/gorilla/rpc/blob/main/v2/protorpc/server.go#L119-L123) where it uses the datatype for Error, but, specifically for v2/json2/server.go, the error key hasn't been specified in [WriteResponse](https://github.com/gorilla/rpc/blob/main/v2/json2/server.go#L179-L183). I am not sure of the exact reason yet, but as it wasn't used anywhere, golangci-lint complained about it, so, I've commented it for the time being.

Comment (coreydaley) [reply v2/json2/server.go:16]: Sounds good.

File: json/client.go (hunk 2 above)
Comment (coreydaley) [reply json/client.go:43]: Yes, i think it should be fine.

**Review by coreydaley — APPROVED** [review]

**Pull request merged**